src/OrchardCore/OrchardCore.Data.YesSql/Removing/ShellDbTablesInfo.cs

```diff
@@ -0,0 +1,331 @@
+using System;
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Linq;
+using Dapper;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using YesSql;
+using YesSql.Services;
+using YesSql.Sql;
+using YesSql.Sql.Schema;
+
+namespace OrchardCore.Environment.Shell.Removing;
+
+internal class ShellDbTablesInfo : ISchemaBuilder
+{
+    private ICommandInterpreter _commandInterpreter;
+    public string DatabaseProvider { get; set; }
+    public string TablePrefix { get; set; }
+    public IEnumerable<string> TableNames { get; set; } = Enumerable.Empty<string>();
+    public ISqlDialect Dialect { get; private set; }
+    public ITableNameConvention TableNameConvention { get; private set; }
+    public DbConnection Connection { get; set; }
+    public DbTransaction Transaction { get; set; }
+    public bool ThrowOnError { get; private set; }
+    public ILogger _logger { get; set; } = NullLogger.Instance;
+
+    public HashSet<(string Name, Type Type, string Collection)> MapIndexTables { get; private set; } =
+        new HashSet<(string Name, Type Type, string Collection)>();
+
+    public HashSet<(string Name, Type Type, string Collection)> ReduceIndexTables { get; private set; } =
+        new HashSet<(string Name, Type Type, string Collection)>();
+
+    public HashSet<string> BridgeTables { get; private set; } = new HashSet<string>();
+    public HashSet<string> DocumentTables { get; private set; } = new HashSet<string>();
+    public HashSet<string> Tables { get; private set; } = new HashSet<string>();
+
+    public ShellDbTablesInfo Configure(ShellSettings shellSettings)
+    {
+        DatabaseProvider = shellSettings["DatabaseProvider"];
+
+        return this;
+    }
+
+    public ShellDbTablesInfo Configure(IConfiguration configuration)
+    {
+        Dialect = configuration.SqlDialect;
+        TablePrefix = configuration.TablePrefix;
```

Comment:
Do we register `IConfiguration` into the IoC? I did not know that we do. I would expect this to be registered in [OrchardCore.Data.YesSql.OrchardCoreBuilderExtensions`](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore/OrchardCore.Data.YesSql/OrchardCoreBuilderExtensions.cs) after evaluation `YesSqlOptions` options. using something like this 

```
services.AddScoped(sp =>
{
    var yesSqlOptions = sp.GetService<IOptions<YesSqlOptions>>().Value;

    return GetStoreConfiguration(sp, yesSqlOptions);
});
```

I suggest that to make sure we honor the `shellSettings["TablePrefixSeparator"]`

Reply:
> Do we register IConfiguration into the IoC?

No, I use somewhere `shellDbTablesInfo.Configure(store.Configuration)`, store being resolved from the DI

Reply:
Here we use configuration from YesSql, we resolve `IStore` that executes the code you mentioned above, then we pass to this `Configure()` method `store.Configuration` wich is a `YesSql.IConfiguration`, not a dotnet `IConfiguration`.

Reply:
Okay. I just wanted to make sure that `YesSql.IConfiguration` resolve the instance that we configured using the `YesSqlOptions`.